types/lodash/common/lang.d.ts

```diff
@@ -574,6 +574,7 @@ declare module "../index" {
         isEmpty<T extends { __trapAny: any }>(value?: T): boolean;
         isEmpty(value: string): value is '';
         isEmpty(value: Map<any, any> | Set<any> | List<any> | null | undefined): boolean;
+        isEmpty(value: Record<string, any>): boolean;
```

Comment:
We may want to do this for all objects, not just records. For example:

```ts
interface Foo {
  bar?: string;
}
const foo: Foo = {};
if (Math.random()) {
  foo.bar = "bar";
}

// Later...
if(_.isEmpty(foo)) {
  foo.bar = "baz"; // This should be allowed
}
```

Reply:
This test case _does_ pass with `Record` type. I cant immediately think of a type that's more explicit that it matches this case.

Reply:
I see - I guess `Record<string, any>` is nearly the same as `object`. I'm not 100% sure what the difference is - maybe `object` includes arrays while  `Record<string, any>` does not. But that doesn't matter since a previous overload already caught arrays.

I think it would be better to replace `Record<string, any>` with `object` here for readability (not a big deal though).